### ISSUE TITLE
InjectionProperty does not work for bool value as expected for Unity v.5.9.0 and later

### ISSUE THREAD

**Issue by hypercodeplace** [unitycontainer/abstractions#107]
### InjectionProperty does not work for bool value as expected for Unity v.5.9.0 and later

The problem you changed the signature of `InjectionProperty` constructor [here](https://github.com/unitycontainer/abstractions/commit/891f76976d094834257eb6adfbeb0cbefea8b4d9#diff-0355ff550f0c041139d4d0c8d0bab177R22). 

It was the following constructors of `InjectionProperty`:

```C#
public InjectionProperty(string name) { ... }                // #1
public InjectionProperty(string name, object value) { ... }  // #2
```

but it were changed in v.5.9.0 to:

```C#
public InjectionProperty(string name, bool optional = false) { ... }  // #1
public InjectionProperty(string name, object value) { ... }           // #2
```

In case `bool` value for version `5.8.13` the constructor `#2` is executed but for version `5.9.0` the constructor `#1` is executed instead. It leads that already existing code that use `InjectionProperty` with `bool` value have changed the behavior. It's very bad but even worse is that there are no unit tests in `Unity` for `InjectionProperty` that failed after these changes and indicated the problem.

### To Reproduce

```C#
public interface IFoo
{
    string StringProperty { get; set; }
    bool BoolProperty { get; set; }
}

public class Foo : IFoo
{
    public string StringProperty { get; set; }
    public bool BoolProperty { get; set; }
}

public class UnityTests
{
    [Test]
    public void TestStringPropertyInjection()
    {
        const string expectedResult = "true";

        var foo = new UnityContainer()
            .RegisterType<IFoo, Foo>(
                new InjectionProperty(nameof(IFoo.StringProperty), expectedResult))
            .Resolve<IFoo>();

        // Passed for 5.8.13 and 5.9.0
        Assert.AreEqual(expectedResult, foo.StringProperty);
    }

    [Test]
    public void TestBoolPropertyInjection()
    {
        const bool expectedResult = true;

        var foo = new UnityContainer()
            .RegisterType<IFoo, Foo>(
                new InjectionProperty(nameof(IFoo.BoolProperty), expectedResult))
            .Resolve<IFoo>();

        // Passed for 5.8.13 and failed for 5.9.0
        Assert.AreEqual(expectedResult, foo.BoolProperty);
    }
}
```

**Comment by hypercodeplace** [comment]
You have fixed wrong behavior for version 5.11.0. But versions 5.9.x and 5.10.x still have wrong behavior. I am still sure this must be noticed at the release notes of version 5.9.0 otherwise everyone who will update (or already updated) the package from version <5.9.0 to these versions have a wrong behavior without any information about that and it "creates significant issue" as you right mentioned.

**Comment by ENikS** [comment]
If you want to create a patch for these versions as well, create relevant pull requests and I will be happy to release it.

Same goes for the release notes
